work_queue/src/work_queue_json.c

```diff
@@ -0,0 +1,284 @@
+            "time_workers_execute_exhaustion", "time_workers_execute_failure", "bytes_received",
+            "bytes_sent", "bytes_transferred", "resources_allocated", "resources_measured", 
+            "resources_requested", "monitor_output_directory", "monitor_snapshot_file", "features",
+            "time_task_submit", "time_task_finish", "time_committed", "time_send_input_start",
+            "time_send_input_finish", "time_receive_result_start", "time_receive_result_finish",
+            "time_receive_output_start", "time_receive_output_finish", "time_execute_cmd_start",
+            "time_execute_cmd_finish", "total_transfer_time", "cmd_execution_time", 
+            "total_cmd_execution_time", "total_cmd_exhausted_execute_time", 
+            "total_time_until_worker_failure", "total_bytes_received", "total_bytes_sent", 
+            "total_bytes_transferred", "time_app_delay"};
+
+
+int is_in(const char* str, char** a)
+{
+
+    char** ptr = a;
+
+    while(*ptr != 0){
+
+        if(!strcmp(*ptr, str)){
+            return 1;
+        }
+
+        ptr++;
+
+    }
+    
+    return 0;
+
+}
+
+int validate_json(struct jx *json, char** a){
+
+    //iterate over the keys in a JX_OBJECT
+    void *j = NULL;
+    const char* key = jx_iterate_keys(json, &j);
+
+    while (key != NULL) {
+
+        if(!is_in(key, a)){
+            return 1;
+        }
+
+        key = jx_iterate_keys(json, &j);
+
+    }
+
+    return 0;
+
+}
+
+int specify_files(int input, struct jx *files, struct work_queue_task *task){
+
+    void *i = NULL;
+    struct jx *arr = jx_iterate_array(files, &i);
+
+    while (arr != NULL){
+        
+        char *local, *remote;
+        int flags;
+        void *k = NULL;
+        void *v = NULL;
+
+        const char *key = jx_iterate_keys(arr, &k);
+        struct jx *value = jx_iterate_values(arr, &v);
+
+        while (key != NULL){
+
+
+            if (!strcmp(key, "local")){
+                local = value->u.string_value;
+            }
+            else if (!strcmp(key, "remote")){
+                remote = value->u.string_value;
+            }
+            else if (!strcmp(key, "flags")){
+                flags = value->u.integer_value;
+            }
+            else{
+                printf("%s\n",value->u.string_value);
+            }
+
+            key = jx_iterate_keys(arr, &k);
+            value = jx_iterate_values(arr, &v);
+
+        }
+
+        if (input){
+            work_queue_task_specify_file(task, local, remote, WORK_QUEUE_INPUT, flags);
+        }
+        else{
+            work_queue_task_specify_file(task, local, remote, WORK_QUEUE_OUTPUT, flags);
+        }
+
+        arr = jx_iterate_array(files, &i);
+    
+    }
+
+    return 0;
+
+}
+
+
+/* WQTCREATE
+ * takes in a JSON string and returns a work_queue struct
+ */
+struct work_queue_task* work_queue_task_json_create(char* str){
+
+    char *command_line;
+    struct jx *input_files, *output_files;
+
+    struct jx *json = jx_parse_string(str); 
```

Comment:
Consider what happens if the user passes in invalid json, and so jx_parse_string returns null.